### PR TITLE
Refactor FXIOS-5008 [v108] Remove SnapKit from InstructionsViewController

### PR DESCRIPTION
**Pull request by thatswinnie**
The InstructionsView is now using SwiftUI.

[Jira](https://mozilla-hub.atlassian.net/browse/FXIOS-5008)
#12035

### CODE REVIEW

File: Client/Frontend/Extensions/DevicePickerViewController.swift [hunk 1]
```diff
@@ -145,9 +175,11 @@ class DevicePickerViewController: UITableViewController {
 
         if !devices.isEmpty {
             if indexPath.section == 0 {
-                cell = tableView.dequeueReusableCell(withIdentifier: DevicePickerTableViewHeaderCell.CellIdentifier, for: indexPath) as! DevicePickerTableViewHeaderCell
+                cell = tableView.dequeueReusableCell(withIdentifier: DevicePickerTableViewHeaderCell.cellIdentifier,
+                                                     for: indexPath) as! DevicePickerTableViewHeaderCell
```
Comment (nbhasin2) [inline Client/Frontend/Extensions/DevicePickerViewController.swift:179]: nit: force unwrap

File: Client/Frontend/DevicePickerTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class DevicePickerTableViewCell: UITableViewCell {
+
+    private struct UX {
+        static let deviceRowTextFont = UIFont.systemFont(ofSize: 16)
+        static let deviceRowTextPaddingLeft: CGFloat = 72
+        static let deviceRowTextPaddingRight: CGFloat = 50
+    }
+
+    static let cellIdentifier = "ClientPickerTableViewCell"
```
Comment (nbhasin2) [inline Client/Frontend/DevicePickerTableViewCell.swift:15]: nit: Could we use @polymathruby's cell identifier extension? 🤔 

Look for `ReusableCell`

File: Client/Frontend/Extensions/DevicePickerViewController.swift [hunk 3]
```diff
@@ -145,9 +175,11 @@ class DevicePickerViewController: UITableViewController {
...
             } else {
-                let clientCell = tableView.dequeueReusableCell(withIdentifier: DevicePickerTableViewCell.CellIdentifier, for: indexPath) as! DevicePickerTableViewCell
+                let clientCell = tableView.dequeueReusableCell(withIdentifier: DevicePickerTableViewCell.cellIdentifier,
```
Comment (nbhasin2) [inline Client/Frontend/Extensions/DevicePickerViewController.swift:181]: nit: force unwrap as well

File: Client/Frontend/HostingTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import SwiftUI
+
+final class HostingTableViewCell<Content: View>: UITableViewCell {
+
+    let cellIdentifier = "HostingTableViewCell"
```
Comment (nbhasin2) [inline Client/Frontend/HostingTableViewCell.swift:9]: nit: I believe we can use ReusableCell here as well 🤞🏼

**Comment by nbhasin2** [comment]
Overall looks good @thatswinnie 👍🏼 
Added some minor nits.

I also tested this on iPhone 14 and iPad Air sim both with and without sync.

**Review by nbhasin2 — APPROVED** [review]

**Review by OrlaM — APPROVED** [review]
LGTM

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**